### DISCUSSION TITLE
How to make a filter choice mandatory?

### DISCUSSION THREAD

**Discussion by cpuiggar** [atoti/atoti#821] [Q&A]
Hey, I have to create a cube, where the measures need to have the filter "ITP" selected in order to show values. Do you know how can I make this possible? Or at least make a filter with a default value? 

Thanks in advance!

**Comment by OPeyrusse** [comment]
Assuming that your filter value "ITP" is on a level, you could wrap your measures with a https://docs.atoti.io/latest/api/atoti.where.html#atoti-where to return only when the value is set.

If your filters are special values, you can always consider placing them on a single-level hierarchy and make this one [slicing](https://docs.atoti.io/latest/api/atoti.Hierarchy.slicing.html#atoti.Hierarchy.slicing). This way, by design, you will have a selected filter by default. If you also set a [custom order](https://docs.atoti.io/latest/api/atoti.CustomOrder.html#atoti.CustomOrder), you can end up with "ITP" being the default filter. Yet, you still need to use a `#where` to properly consider the filter.

There are likely other alternatives but it will require that you post some code or sample project to better understand the context in which you are trying to make this work :smile:
